File: backend/app/models/game.py
from typing import List, Optional
from pydantic import BaseModel, Field


class GameState(BaseModel):
    """ゲーム状態"""
    session_id: str = Field(..., description="セッションID")
    observation: str = Field(..., description="現在の観察結果")
    available_actions: List[str] = Field(..., description="利用可能なアクション")
    score: int = Field(default=0, description="現在のスコア")
    reward: Optional[int] = Field(None, description="最後のアクションの報酬")
    done: bool = Field(default=False, description="ゲーム終了フラグ")
    max_steps: int = Field(default=100, description="最大ステップ数")
    current_step: int = Field(default=0, description="現在のステップ数")


class GameStep(BaseModel):
    """ゲームステップの履歴"""
    step_number: int
    action: str
    observation: str
    reward: int
    score: int
    done: bool


class GameSession(BaseModel):
    """ゲームセッション"""
    session_id: str
    game_id: str
    created_at: str
    last_accessed: str
    current_step: int = 0
    history: List[GameStep] = []
    
    class Config:
        arbitrary_types_allowed = True

File: frontend/src/types/game.ts
export interface GameLog {
  id: string;
  action: string;
  observation: string;
  score: number;
  reward: number;
  done: boolean;
  timestamp: number;
  available_actions?: string[];
}

export interface GameState {
  currentGame: string | null;
  mode: 'manual' | 'auto' | null;
  logs: GameLog[];
  isProcessing: boolean;
  autoRunning: boolean;
  currentStep: number;
  totalReward: number;
  isDone: boolean;
}

export const GAMES = [
  {
    id: 'treasure-hunt',
    name: 'Treasure Hunt',
    difficulty: 'Medium',
    description: 'Find the hidden treasure in a mysterious mansion filled with puzzles and secrets.',
    maxSteps: 50,
  },
];

export const AI_AGENT_STATES = {
  idle: {
    message: "準備OK。短文で誘導してね。",
    mood: "😊",
  },
  thinking: {
    message: "考え中…",
    mood: "🤔",
  },
  success: {
    message: "実行しました！",
    mood: "✨",
  },
  error: {
    message: "うまくいかなかった…言い換えてみて",
    mood: "😅",
  },
  auto: {
    message: "自動進行中…",
    mood: "🔄",
  },
};
